metricflow/model/parsing/dir_to_model.py

```diff
@@ -220,38 +219,47 @@ def parse_config_yaml(
             document_type = next(iter(config_document.keys()))
             object_cfg = config_document[document_type]
 
-            if document_type == METRIC_TYPE:
-                metric_validator.validate(config_document[document_type])
-                results.append(metric_class.parse_obj(object_cfg))
-            elif document_type == DATA_SOURCE_TYPE:
-                data_source_validator.validate(config_document[document_type])
-                results.append(data_source_class.parse_obj(object_cfg))
-            elif document_type == MATERIALIZATION_TYPE:
-                materialization_validator.validate(config_document[document_type])
-                results.append(materialization_class.parse_obj(object_cfg))
-            else:
+            try:
+                if document_type == METRIC_TYPE:
+                    metric_validator.validate(config_document[document_type])
+                    results.append(metric_class.parse_obj(object_cfg))
+                elif document_type == DATA_SOURCE_TYPE:
+                    data_source_validator.validate(config_document[document_type])
+                    results.append(data_source_class.parse_obj(object_cfg))
+                elif document_type == MATERIALIZATION_TYPE:
+                    materialization_validator.validate(config_document[document_type])
+                    results.append(materialization_class.parse_obj(object_cfg))
+                else:
+                    issues.append(
+                        ValidationError(
+                            context=FileContext(file_name=ctx.filename, line_number=ctx.start_line),
+                            message=f"Invalid document type: {document_type}. Expected {DOCUMENT_TYPES}.",
+                        )
+                    )
+            # catches exceptions from jsonschema validator
+            except exceptions.ValidationError as e:
```

Comment:
Oh I see, this thing is from jsonschema. Got it.

Reply:
Yea 🙃 I should have realized names like `ValidationError` would eventually have collisions like this. But I didn't, and so here we are 😬